File: backend/KKUKKKUK/src/main/java/com/be/KKUKKKUK/domain/diagnosis/entity/Diagnosis.java
package com.be.KKUKKKUK.domain.diagnosis.entity;

import com.be.KKUKKKUK.domain.hospital.entity.Hospital;
import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.*;

/**
 * packageName    : com.be.KKUKKKUK.domain.diagnosis.entity<br>
 * fileName       : Diagnosis.java<br>
 * author         : eunchang <br>
 * date           : 2025-04-07<br>
 * description    : Diagnosis의 entity 클래스입니다.  <br>
 * ===========================================================<br>
 * DATE              AUTHOR             NOTE<br>
 * -----------------------------------------------------------<br>
 * 25.04.07          eunchang           최초생성<br>
 * <br>
 */

@Entity
@Table(name = "diagnosis")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Diagnosis {
    @Id
    @Column(name="id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "hospital_id", nullable = false )
    private Hospital hospital;

    @Column(name = "name", length = 100, nullable = false)
    private String name;

    @Builder
    public Diagnosis(String name, Hospital hospital){
        this.name = name;
        this.hospital = hospital;
    }
}

File: frontend/src/interfaces/Diagnosis.ts
import { ApiResponse } from "@/types";

/**
 * @module Diagnosis
 * @file Diagnosis.ts
 * @author eunchang
 * @date 2025-03-26
 * @description 검사 관련 이터페이스 모듈입니다.
 *
 * 검사 관련 인터페이스를 모아서 관리합니다.
 *
 * ===========================================================
 * DATE              AUTHOR             NOTE
 * -----------------------------------------------------------
 * 2025-04-09        eunchang         Diagnosis 추가
 */

/**
 * 진단 데이터 인터페이스
 *
 * @interface
 */
export interface Diagnosis {
  id: number;
  name: string;
  hospitalId: string;
}

/**
 * 단일 진단 응답 인터페이스
 *
 * @interface
 */
export interface DiagnosisResponse extends ApiResponse<Diagnosis | null> {}

/**
 * 진단목록 응답 인터페이스
 *
 * @interface
 */
export interface DiagnosisListResponse extends ApiResponse<Diagnosis[]> {}

/**
 * 자동완성 응답 인터페이스
 *
 * @interface
 */
export interface AutoCorrectDiagnosisResponse extends ApiResponse<string[]> {}
/**
 * 검사 생성/수정 요청 인터페이스
 *
 * @interface
 */
export interface DiagnosisRequest {
  name: string;
}
